### ISSUE TITLE
One specific BQ-DBT SQL file gets wrongly formatted and becomes unusable

### ISSUE THREAD

**Issue by tgmof** [AltimateAI/vscode-dbt-power-user#856]
### Summary

I am considering using sqlfmt and found out that when I format it with sqlfmt, it makes it unusable because it accidentally comments out code that was in used.
This is not a critical bug since playing around with the file before running sqlfmt solves the issue.
I also created an issue in sqlfmt repo but I actually can't reproduce it with their command line tool. It's only when I run it through VSCode following your documentation on sqlfmt.
I will send you the file via the Chat because I cannot share it publicly.
I think you are simply using sqlfmt with different parameter as what I'm using, but I can't see which parameter you add and thus I cannot raise the issue to them "properly":
https://github.com/tconbeer/sqlfmt/issues/554

### Expected behavior

Expected: The file is formatted and the code is still valid.

### Actual behavior

The DBT-jinja code is invalid.

### Steps To Reproduce

Put the file I sent in a VSCode repo.
Notice that the comments from line 99 to 104 use the {##} syntax to have DBT-jinja comments that are not rendered in the final SQL code.
Setup "dbt.sqlFmtAdditionalParams": ["-l", "120"]
Right-click: Format Document.
Notice that the comment mentioned above now extended from line 74 to line 104 and thus breaking the code.

### Operating System

Ubuntu WSL (Linux VM in Windows)

### dbt version

1.5.1

### dbt Adapter

BigQuery

### dbt Power User version

v0.30.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by mdesmet** [comment]
Is this issue an extension issue or can it be simulated by running sqlfmt directly?

**Comment by tconbeer** [comment]
OP Mentioned in the other issue that they couldn't reproduce with sqlfmt directly, but I don't know exactly what that means.

@tgmof can you send me the file through the dbt slack, or just change/obfuscate the sensitive names and post it here?

**Comment by tgmof** [comment]
@tconbeer I was able to reproduce with a simpler file given from the dbt init that I modified to be in a case where the formatter has to delete of lot of line. sqlfmt works as expected but VSCode Extension makes a wrong insert and makes the query invalid.
Trying to debug further now but I think it comes from the fact that processDiffOutput only take care of the diff that are output by sqlfmt but doesn't do anything to the thing "in-between chunks". that are not output (because already correct) by sqlfmt. So the things "in-between chunks" basically don't move even though they should be impacted by the previous chunk.

/*
    Welcome to your first dbt model!
    Did you know that you can also configure models directly within SQL files?

    This will override configurations stated in dbt_project.yml
    Try changing "table" to "view" below
*/
{{ config(materialized="table") }}

with

    source_data as (

        select 1 as id











        union all select null as id

    )

select *
from source_data

union all
select *
from 







source_data

    /*
    Uncomment the line below to remove records with null `id` values
*/
    -- where id is not null
